src/pages/build-apps/examples/heystack.md

```diff
@@ -0,0 +1,164 @@
+
+- Authenticating users with the web wallet
+- Using a smart contract to store data on the blockchain
+- Minting new fungible tokens with a SIP-010 compliant smart contract
+- Creating and monitoring transactions on the Stacks blockchain using Stacks.js
+
+You can access the [online version][heystack] of the Heystack app to interact with it. The source for Heystack is also
+available on [Github][heystack_gh]. This page assumes that you are familiar with [React][].
+
+## Review smart contracts
+
+Heystack depends on two smart contracts to execute the backend functions of the app on the Stacks blockchain: a contract
+for handling the messaging content, and a contract for minting and distributing the $HEY token.
+
+### Content contract
+
+The `hey.clar` contract provides two primary functions for the application, a function to publish content to
+the blockchain and a function to like a piece of content based on its ID. This section reviews the implementation of
+these primary functions, but is not a comprehensive discussion of the contract.
+
+In order to accomplish the two primary functions, the contract relies on a data variable `content-index` and two
+[data maps][], `like-state` and `publisher-state` which contain the number of likes a piece of content has received, and
+the principal address of the account that published the content.
+
+Note that all variables are defined at the top of the contract, which is a requirement of the Clarity language. These
+include constants such as the `contract-creator`, error codes, and a treasury address.
+
+```clarity
+;;
+;; Data maps and vars
+(define-data-var content-index uint u0)
+
+(define-read-only (get-content-index)
+  (ok (var-get content-index))
+)
+
+(define-map like-state
+  { content-index: uint }
+  { likes: uint }
+)
+
+(define-map publisher-state
+  { content-index: uint }
+  { publisher: principal }
+)
+```
+
+Read-only functions provide a method for getting the like count of a piece of content, and getting the principal address
+of the message publisher.
+
+```clarity
+(define-read-only (get-like-count (id uint))
+  ;; Checks map for like count of given id
+  ;; defaults to 0 likes if no entry found
+  (ok (default-to { likes: u0 } (map-get? like-state { content-index: id })))
+)
+
+(define-read-only (get-message-publisher (id uint))
+  ;; Checks map for like count of given id
+  ;; defaults to 0 likes if no entry found
+  (ok (unwrap-panic (get publisher (map-get? publisher-state { content-index: id }))))
+```
+
+The `get-like-count` method accepts a content ID and returns the number of likes associated with that content. The
+method uses the [`default-to`][] function to return `0` if the content ID isn't found in the map of likes.
+
+The `get-message-publisher` method accepts a content ID and returns the principal address of the content publisher. The
+method uses the [`unwrap-panic`][] function to halt execution of the method if the principal address isn't found in
```

Comment:
I think this method just pulls the values out of the response, and fails if that doesn't work, rather than anything map specific @lgalabru @reedrosenbluth would be best to clarify most accurate phrasing